pkg/iplister/decoder/file/file.go

```diff
@@ -0,0 +1,47 @@
+package file
+
+import (
+	"io"
+
+	"gopkg.in/yaml.v3"
+)
+
+type File struct {
+	sources []string
+}
+
+// Structs for marshalling in data from an ipLists file
+type (
+	ipLists struct {
+		IPLists map[string][]string `json:"ipLists"`
+	}
+)
+
+func New(sources ...string) *File {
```

Comment:
What is the benefit of providing the `sources`  within the constructor instead of resolving them from the file that is Decoded?

Reply:
I was thinking about the consumer of this. The file will contain different sources, it may only need to get the "jira" IPs. Having the `sources` in the constructor allows `GetIPs()` to filter the results while still maintaining the current no-arg interface.

```
        ipReader := file.New("iplists.yaml")
        jiraIPDecoder := filedeocder.New("jira") // instantiate here

	lister := iplister.New(ipReader, jiraIPDecoder)
	jiraIPs, err := lister.GetIPs() // no args needed here still
```

Reply:
I see.  That example was very helpful!